rusk/src/lib/chain/rusk.rs

```diff
@@ -147,9 +148,14 @@ impl Rusk {
                     block_gas_left -= gas_spent;
                     let gas_price = unspent_tx.inner.fee.gas_price;
                     dusk_spent += gas_spent * gas_price;
+                    let economic_gas_spent = match receipt.economic_mode {
+                        Allowance(allowance) if allowance > 0 => 0,
+                        Charge(charge) => gas_spent + charge,
+                        _ => gas_spent,
+                    };
                     spent_txs.push(SpentTransaction {
                         inner: unspent_tx,
-                        gas_spent,
+                        gas_spent: economic_gas_spent,
```

Comment:
Please add the economic mode to a different field
Don't mix the gasSpent with additional charge

Doing that, you'll probably have to change how spent transactions are serialize and expose those new fields in the gql operations

Reply:
As agreed offline, we'll wait for the feedback of others as this is a more general issue

Reply:
I think the removal of the surcharge will mitigate this problem, however not totally remove it. I agree with @herr-seppia on this. There should be a different field, telling us how much has been allowed/charged by the contract.

If nothing else, this will allow us to display this working downstream (explorer), and it can always be used by rusk itself to compute the refund.

Reply:
Regardless of the estimated surcharge, I also think these should be two separate fields and have Rusk decide payings and refunds.

Reply:
the above issues have now been addressed - both surcharge as well as a separate field in spent transaction for the economic gas spent - please have a look